### PR TITLE
updated to wpi 2023 and started april tags a little bit

### PR DESCRIPTION
**Pull request by pixelmonaskarion**

### CODE REVIEW

File: src/main/java/frc/robot/Robot.java [hunk 1]
```diff
@@ -1,75 +1,100 @@
...
+	@Override
+	public void robotInit() {
+		m_robotContainer = new RobotContainer();
+		// register subsystems into a list to minimize redundancy if logging
+		// inputs/outputs for all
+		camera = new PhotonCamera("USB2.0_PC_CAMERA");
+	}
+
+	@Override
+	public void robotPeriodic() {
```
Comment (Irishumanoid) [inline src/main/java/frc/robot/Robot.java:37]: Can someone confirm this has been tested and is functional before it is merged into main?

File: src/test/java/WaypointParserTest.java [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
     private TrajectoryGenerator.ControlVectorList controlVectors = new TrajectoryGenerator.ControlVectorList();
     
     @Test
     public void getWaypoints() throws IOException { 
-    controlVectors.add(new Spline.ControlVector(new double[] {7.370389231190149, -0.6245939917920662}, new double[] {-3.542893461012311+8.1, 0.23011357592339232}));
-    controlVectors.add(new Spline.ControlVector(new double[] {5.874650987688097, -0.8194487104192959}, new double[] {-4.430474396716827+8.1, -0.8743028395305534}));
-    controlVectors.add(new Spline.ControlVector(new double[] {5.052816787961696, 0.18080352393980803}, new double[] {-6.2878196880984945+8.1, -0.7889608317373478}));
-
-    assertEquals(WaypointParser.getControlVectors(Filesystem.getDeployDirectory().toPath().resolve("/src/main/deploy/Pathweaver/paths/exit1.path")), controlVectors);
+        var expectedList = new ArrayList<List<Double>>();
```
Comment (Irishumanoid) [inline src/test/java/WaypointParserTest.java:20]: why are you instantiating this?

File: src/main/java/frc/robot/Robot.java (hunk 1 above)
Comment (pixelmonaskarion) [reply src/main/java/frc/robot/Robot.java:37]: It has been tested, it doesn't break anything but it doesn't work quite yet either

Comment (Irishumanoid) [reply src/main/java/frc/robot/Robot.java:37]: then let's wait until tomorrow to debug and then merge

File: src/test/java/WaypointParserTest.java (hunk 2 above)
Comment (pixelmonaskarion) [reply src/test/java/WaypointParserTest.java:20]: I changed your test code a bit, fixed file loading. the problem right now is that you cannot compare with that datatype, I was starting to fix that with that line

**Review by Irishumanoid — APPROVED** [review]

File: src/test/java/WaypointParserTest.java [hunk 3]
```diff
@@ -1,23 +1,28 @@
...
+        controlVectors.add(new Spline.ControlVector(new double[] {7.370389231190149, -0.6245939917920662}, new double[] {-3.542893461012311+8.1, 0.23011357592339232}));
+        controlVectors.add(new Spline.ControlVector(new double[] {5.874650987688097, -0.8194487104192959}, new double[] {-4.430474396716827+8.1, -0.8743028395305534}));
+        controlVectors.add(new Spline.ControlVector(new double[] {5.052816787961696, 0.18080352393980803}, new double[] {-6.2878196880984945+8.1, -0.7889608317373478}));
+        var pathweaver_file = FileSystems.getDefault().getPath("src/main/deploy/Pathweaver/paths/exit1.path");
+        var parsedwaypoints = WaypointParser.getControlVectors(pathweaver_file);
+        //assertEquals(controlVectors, parsedwaypoints);
```
Comment (garrettwitowski) [inline src/test/java/WaypointParserTest.java:26]: As a rule of thumb we should not be submitting/merging commented out code, as this reduces readability and can be confusing to others. Also, any unit tests should be fully functional prior to submitting/merging.

**Pull request merged**